web-common/src/features/dashboards/proto-state/fromProto.ts

```diff
@@ -160,30 +160,30 @@ export function base64ToProto(message: string) {
   return protoBase64.dec(message);
 }
 
-function fromExpressionProto(expression: Expression) {
+function fromExpressionProto(expression: Expression): V1Expression | undefined {
   switch (expression.expression.case) {
     case "ident":
       return {
         ident: expression.expression.value,
-      } as V1Expression;
+      };
 
     case "val":
       return {
         val:
           expression.expression.value.kind.case === "nullValue"
             ? null
             : expression.expression.value.kind.value,
-      } as V1Expression;
+      };
 
     case "cond":
       return {
         cond: {
           op: FromProtoOperationMap[expression.expression.value.op],
-          exprs: expression.expression.value.exprs.map((e) =>
-            fromExpressionProto(e),
-          ),
+          exprs: expression.expression.value.exprs
+            .map((e) => fromExpressionProto(e))
+            .filter((e) => e !== undefined) as V1Expression[],
```

Comment:
Here's a little trick for filters like this @AdityaHegde -- did you know that you can use:
`.filter((e): e is V1Expression => e !== undefined)`
to avoid the cast? It basically asserts that if `e is V1Expression` then the predicate must be true, and then TS is smart enough to propagate the type! I only just learned this the other day myself (I have code that I'll need to clean up to use this), so I thought I'd share it :-)